### PR TITLE
Update bazel documentation with current known issues and limitations

### PR DESCRIPTION
**Pull request by ixdy**
@liztio asked about this last month, which made me realize this was a bit out of date.

cc @fejta @BenTheElder @jayconrod

### CODE REVIEW

File: contributors/devel/bazel.md [hunk 1]
```diff
@@ -90,6 +79,92 @@ Updating the `BUILD` file for a package will be required when:
...
+from an amd64 machine.)
+
+The Go rules support cross-compilation of pure Go code using the `--platforms`
+flag, and this is being used successfully in the kubernetes/test-infra repo.
+
+It may already be possible to cross-compile cgo code if a custom CC toolchain is
+set up, possibly reusing the kube-cross Docker image, but this area needs
+further exploration.
+
+### The CC toolchain is not fully hermitic
```
Comment (BenTheElder) [inline contributors/devel/bazel.md]: should we mention `bazel clean --expunge` when you upgrade these?

Comment (ixdy) [reply contributors/devel/bazel.md]: good suggestion, done

**Comment by liztio** [comment]
thank you @ixdy!

**Pull request merged**